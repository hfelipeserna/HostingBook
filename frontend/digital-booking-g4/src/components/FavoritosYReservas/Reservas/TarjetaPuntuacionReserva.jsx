import styles from "../../Producto/ProductoReserva/DetallesReserva/TarjetaReservaExitosa/TarjetaReservaExitosa.module.css";
import FilledButton from "../../Buttons/FilledButton";
import { useHistory } from "react-router";

export default function TarjetaPuntuacionReserva({ nombreAlojamiento }) {
  const history = useHistory();

  return (
    <div className={styles.card}>
      
      <h2 className={`${styles.title} ${styles.calificarTitle}`}>
        Calificá tu experiencia en {nombreAlojamiento}!
      </h2>

      <div className={styles.starWidget}>
        <input type="radio" name="rate" id="rate5" />
        <label htmlFor="rate5" className="fas fa-star"></label>
        <input type="radio" name="rate" id="rate4" />
        <label htmlFor="rate4" className="fas fa-star"></label>
        <input type="radio" name="rate" id="rate3" />
        <label htmlFor="rate3" className="fas fa-star"></label>
        <input type="radio" name="rate" id="rate2" />
        <label htmlFor="rate2" className="fas fa-star"></label>
        <input type="radio" name="rate" id="rate1" />
        <label htmlFor="rate1" className="fas fa-star"></label>
      </div>

      <FilledButton onClick={() => console.log("Envio calificación..")} styles={styles.btnOK}>
        Confirmar
      </FilledButton>
    </div>
  );
}
