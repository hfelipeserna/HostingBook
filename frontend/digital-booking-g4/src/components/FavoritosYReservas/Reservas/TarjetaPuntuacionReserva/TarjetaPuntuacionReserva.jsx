import styles from "../../../Producto/ProductoReserva/DetallesReserva/TarjetaReservaExitosa/TarjetaReservaExitosa.module.css";
import FilledButton from "../../../Buttons/FilledButton";
import { useState } from "react";
import post from "../../../../utils/post";

export default function TarjetaPuntuacionReserva({
  nombreAlojamiento,
  idAlojamiento,
}) {
  const [puntuacion, setPuntuacion] = useState(null);
  const idUsuario = parseInt(localStorage.getItem("id"));
  const [commentText, setCommentText] = useState(""); 
  const [puntuadoConExito, setPuntuadoConExito] = useState(null);

  console.log(puntuacion);

  const enviarPuntuacion = () => {
    post("puntuaciones", {
      puntuacion: puntuacion,
      comentario: commentText,
      producto: { id: idAlojamiento },
      usuario: { id: idUsuario },
    })
      .then((response) => {
        console.log(response.status);
        if (response.status !== 200) {
          setPuntuadoConExito(false);
        } else {
          setPuntuadoConExito(true);
        }
      })
      .catch((err) => console.log(err));
  };

  return (
    <div className={styles.card}>
      {puntuadoConExito === null ? (
        <>
          <h2 className={`${styles.title} ${styles.calificarTitle}`}>
            Calificá tu experiencia en {nombreAlojamiento}!
          </h2>

          <div className={styles.starWidget}>
            <input
              type="radio"
              name="rate"
              id="rate5"
              value={5}
              onClick={() => setPuntuacion(5)}
            />
            <label htmlFor="rate5" className="fas fa-star"></label>
            <input
              type="radio"
              name="rate"
              id="rate4"
              value={4}
              onClick={() => setPuntuacion(4)}
            />
            <label htmlFor="rate4" className="fas fa-star"></label>
            <input
              type="radio"
              name="rate"
              id="rate3"
              value={3}
              onClick={() => setPuntuacion(3)}
            />
            <label htmlFor="rate3" className="fas fa-star"></label>
            <input
              type="radio"
              name="rate"
              id="rate2"
              value={2}
              onClick={() => setPuntuacion(2)}
            />
            <label htmlFor="rate2" className="fas fa-star"></label>
            <input
              type="radio"
              name="rate"
              id="rate1"
              value={1}
              onClick={() => setPuntuacion(1)}
            />
            <label htmlFor="rate1" className="fas fa-star"></label>
          </div>
          <textarea className={styles.puntuacionComentarios}
            name="comentario"
            id="comentario"
            value={commentText}
            onChange={(e) => setCommentText(e.target.value)}
            placeholder="Contanos tu experiencia (Opcional)"
          ></textarea>
          <FilledButton onClick={enviarPuntuacion} styles={styles.btnOK}>
            Confirmar
          </FilledButton>
        </>
      ) : puntuadoConExito ? (
        <>
          <p className={styles.msjPuntuacionExitosa}>
            Puntuación realizada con éxito!
          </p>
          <i
            className={`far fa-check-circle ${styles.iconoPuntuacionExitosa}`}
          ></i>
        </>
      ) : (
        <>
          <p className={styles.msjPuntuacionFallida}>Puntuación fallida!</p>
          <i
            className={`far fa-times-circle ${styles.iconoPuntuacionFallida}`}
          ></i>
        </>
      )}
    </div>
  );
}
