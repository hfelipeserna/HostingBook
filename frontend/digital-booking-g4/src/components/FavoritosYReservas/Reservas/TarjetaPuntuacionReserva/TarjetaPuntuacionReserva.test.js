import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';

import Router from "react-router-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import { configure, mount } from "enzyme";
import TarjetaPuntuacionReserva from "./TarjetaPuntuacionReserva";
configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe(" Tarjeta Puntuacion Reserva test", function () {
    const nombreAlojamiento = "MeliÃ¡"
    const idAlojamiento = 1
    test("Renderiza Default", () => {
        const renderizacionSimple = render(<TarjetaPuntuacionReserva nombreAlojamiento={nombreAlojamiento} idAlojamiento={idAlojamiento}></TarjetaPuntuacionReserva>)
        expect(renderizacionSimple).toMatchSnapshot()
    })
    test("Reacciona Ante On Change", () => {
        const renderizacionSimple = mount(<TarjetaPuntuacionReserva nombreAlojamiento={nombreAlojamiento} idAlojamiento={idAlojamiento}></TarjetaPuntuacionReserva>)
        const textArea = renderizacionSimple.find("textarea")

        textArea.simulate('change', { target: { value: '   a' } })
        renderizacionSimple.update()


    })
    test("Reacciona Ante On Click", () => {
        const renderizacionSimple = mount(<TarjetaPuntuacionReserva nombreAlojamiento={nombreAlojamiento} idAlojamiento={idAlojamiento}></TarjetaPuntuacionReserva>)

        const input1 = renderizacionSimple.find("input").filter({ id: "rate1" })
        const input2 = renderizacionSimple.find("input").filter({ id: "rate2" })
        const input3 = renderizacionSimple.find("input").filter({ id: "rate3" })
        const input4 = renderizacionSimple.find("input").filter({ id: "rate4" })
        const input5 = renderizacionSimple.find("input").filter({ id: "rate5" })
        input1.simulate('click')
        input2.simulate('click')
        input3.simulate('click')
        input4.simulate('click')
        input5.simulate('click')
        const filledButton = renderizacionSimple.find(".btnOK")
        const form = renderizacionSimple.find("form")
        filledButton.simulate('click')
        filledButton.simulate('submit')
        form.simulate('submit')

        renderizacionSimple.update()


    })

})