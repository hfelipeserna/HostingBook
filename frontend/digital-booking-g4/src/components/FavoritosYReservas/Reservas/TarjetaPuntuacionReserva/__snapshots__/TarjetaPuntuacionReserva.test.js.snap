// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Tarjeta Puntuacion Reserva test Renderiza Default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="card"
      >
        <h2
          class="title calificarTitle"
        >
          Calific치 tu experiencia en 
          Meli치
          !
        </h2>
        <div
          class="starWidget"
        >
          <input
            id="rate5"
            name="rate"
            type="radio"
            value="5"
          />
          <label
            class="fas fa-star estrellaPuntuacion"
            for="rate5"
          />
          <input
            id="rate4"
            name="rate"
            type="radio"
            value="4"
          />
          <label
            class="fas fa-star estrellaPuntuacion"
            for="rate4"
          />
          <input
            id="rate3"
            name="rate"
            type="radio"
            value="3"
          />
          <label
            class="fas fa-star estrellaPuntuacion"
            for="rate3"
          />
          <input
            id="rate2"
            name="rate"
            type="radio"
            value="2"
          />
          <label
            class="fas fa-star estrellaPuntuacion"
            for="rate2"
          />
          <input
            id="rate1"
            name="rate"
            type="radio"
            value="1"
          />
          <label
            class="fas fa-star estrellaPuntuacion"
            for="rate1"
          />
        </div>
        <textarea
          class="puntuacionComentarios"
          id="comentario"
          name="comentario"
          placeholder="Contanos tu experiencia (Opcional)"
        />
        <input
          class="filledButton btnOK"
          type="submit"
          value="Confirmar"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="card"
    >
      <h2
        class="title calificarTitle"
      >
        Calific치 tu experiencia en 
        Meli치
        !
      </h2>
      <div
        class="starWidget"
      >
        <input
          id="rate5"
          name="rate"
          type="radio"
          value="5"
        />
        <label
          class="fas fa-star estrellaPuntuacion"
          for="rate5"
        />
        <input
          id="rate4"
          name="rate"
          type="radio"
          value="4"
        />
        <label
          class="fas fa-star estrellaPuntuacion"
          for="rate4"
        />
        <input
          id="rate3"
          name="rate"
          type="radio"
          value="3"
        />
        <label
          class="fas fa-star estrellaPuntuacion"
          for="rate3"
        />
        <input
          id="rate2"
          name="rate"
          type="radio"
          value="2"
        />
        <label
          class="fas fa-star estrellaPuntuacion"
          for="rate2"
        />
        <input
          id="rate1"
          name="rate"
          type="radio"
          value="1"
        />
        <label
          class="fas fa-star estrellaPuntuacion"
          for="rate1"
        />
      </div>
      <textarea
        class="puntuacionComentarios"
        id="comentario"
        name="comentario"
        placeholder="Contanos tu experiencia (Opcional)"
      />
      <input
        class="filledButton btnOK"
        type="submit"
        value="Confirmar"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
